apps/bfd-pipeline/bfd-pipeline-shared-utils/src/main/java/gov/cms/bfd/pipeline/sharedutils/jobs/store/PipelineJobRecordStore.java

```diff
@@ -0,0 +1,206 @@
+package gov.cms.bfd.pipeline.sharedutils.jobs.store;
+
+import gov.cms.bfd.pipeline.sharedutils.PipelineJob;
+import gov.cms.bfd.pipeline.sharedutils.PipelineJobArguments;
+import gov.cms.bfd.pipeline.sharedutils.PipelineJobOutcome;
+import gov.cms.bfd.pipeline.sharedutils.PipelineJobRecordId;
+import gov.cms.bfd.pipeline.sharedutils.PipelineJobType;
+import java.time.Instant;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Stores and manages the {@link PipelineJobRecord}s representing the history of submitted {@link
+ * PipelineJob}s.
+ */
+public final class PipelineJobRecordStore {
+  private static final Logger LOGGER = LoggerFactory.getLogger(PipelineJobRecordStore.class);
+
+  /** The number of milliseconds to wait between polling job dependencies' status. */
+  private static final int JOB_DEPENDENCY_POLL_MILLIS = 100;
+
+  private final Map<PipelineJobRecordId, PipelineJobRecord<?>> jobRecords;
+
+  /** Constructs a new {@link PipelineJobRecordStore} instance. */
+  public PipelineJobRecordStore() {
+    this.jobRecords = new HashMap<>();
```

Comment:
jobRecords needs to be ConcurrentHashMap since we are using stream() in various methods.  Raw HashMap will throw ConcurrentModificationException if anything is added to or removed from the map during iteration/streaming over it.

Reply:
My brain's a bit fried after the morning's work, so I'll need a break before I try to reason through this all the way, but at first glance, I think you're right. Thanks!

Reply:
Yup: you were correct. Fixed. Thanks!